src/OpenIddict.Core/Infrastructure/OpenIddictProvider.Authentication.cs

```diff
@@ -213,7 +316,56 @@ public partial class OpenIddictProvider<TUser, TApplication, TAuthorization, TSc
                 return;
             }
 
+            if (context.HttpContext.Request.Path == context.Options.AuthorizationEndpointPath &&
+                string.Equals(context.HttpContext.Request.Method, "POST", StringComparison.OrdinalIgnoreCase)) {
+                // Ensure session support has been enabled for this request.
+                if (context.HttpContext.Features.Get<ISessionFeature>() == null) {
+                    services.Logger.LogError("The authorization request was rejected because the session middleware " +
+                                             "was not correctly registered. Session support must be enabled to allow " +
+                                             "processing POST authorization requests.");
+
+                    context.Reject(
+                        error: OpenIdConnectConstants.Errors.InvalidRequest,
+                        description: "POST authorization requests are not allowed by this authorization server.");
+
+                    return;
+                }
+
+                // Generate a request identifier. Note: using a crypto-secure
+                // random number generator is not necessary in this case.
+                var identifier = Guid.NewGuid().ToString();
+
+                // Load the session from the session store.
+                await context.HttpContext.Session.LoadAsync();
+
+                // Store the serialized authorization request parameters in the user session.
+                context.HttpContext.Session.Set("openiddict-request:" + identifier, context.Request.Export());
```

Comment:
`OpenidDictRequestKey = "openiddict-request:"` ? I'm not a huge fan of strings :)